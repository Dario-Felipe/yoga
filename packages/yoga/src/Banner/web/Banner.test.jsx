import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { ThemeProvider, Banner } from '../..';

describe('<Banner />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <Banner message="Feedback message." />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with custom variant', () => {
    const { container } = render(
      <ThemeProvider>
        <Banner variant="success" message="Success banner without button" />
        <Banner
          variant="informative"
          message="Informative banner without button"
        />
        <Banner variant="attention" message="Attention banner without button" />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with button', () => {
    const { container } = render(
      <ThemeProvider>
        <Banner
          message="Banner with button"
          actionLabel="Action"
          onAction={() => {}}
        />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should call button action function when the banner button is clicked', () => {
    const onButtonClickMock = jest.fn();
    const { getByText } = render(
      <ThemeProvider>
        <Banner
          message="Banner with button"
          actionLabel="Action"
          onAction={onButtonClickMock}
        />
      </ThemeProvider>,
    );

    fireEvent.click(getByText('Action'));

    expect(onButtonClickMock).toHaveBeenCalled();
  });
});
