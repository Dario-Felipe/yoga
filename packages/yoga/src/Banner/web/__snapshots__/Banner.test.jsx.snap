// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should match snapshot 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #231B22;
  margin-top: 4px;
  margin-bottom: 4px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  background-color: #E0DFFF;
  padding: 12px 16px;
  border-radius: 8px;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
    >
      Feedback message.
    </p>
  </div>
</div>
`;

exports[`<Banner /> should match snapshot with button 1`] = `
.c2 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 32px;
  padding-left: 16px;
  padding-right: 16px;
  background-color: #231B22;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 14px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 svg {
  width: 16px;
  height: 16px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c2:not([disabled]):hover,
.c2:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(35,27,34,0.45);
}

.c2:active {
  background-color: rgba(35,27,34,0.75);
  color: #FFFFFF;
}

.c2:active svg {
  fill: #FFFFFF;
}

.c3 {
  background-color: transparent;
  border-color: transparent;
  color: #231B22;
}

.c3 svg {
  fill: #231B22;
}

.c3:not([disabled]):hover,
.c3:not([disabled]):focus,
.c3:not([disabled]):active {
  background-color: transparent;
  box-shadow: none;
}

.c3:not([disabled]):hover {
  color: rgba(35,27,34,0.5);
}

.c3:not([disabled]):hover svg {
  fill: rgba(35,27,34,0.5);
}

.c3:not([disabled]):focus,
.c3:not([disabled]):active {
  color: rgba(35,27,34,0.75);
}

.c3:not([disabled]):focus svg,
.c3:not([disabled]):active svg {
  fill: rgba(35,27,34,0.75);
}

.c3:disabled {
  background-color: transparent;
  border-color: transparent;
  color: #D7D7E0;
}

.c3:disabled svg {
  fill: #D7D7E0;
}

.c4 {
  margin-left: 8px;
}

.c1 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #231B22;
  margin-top: 4px;
  margin-bottom: 4px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  background-color: #E0DFFF;
  padding: 12px 16px;
  border-radius: 8px;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
    >
      Banner with button
    </p>
    <button
      class="c2 c3 c4"
    >
      Action
    </button>
  </div>
</div>
`;

exports[`<Banner /> should match snapshot with custom variant 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #231B22;
  margin-top: 4px;
  margin-bottom: 4px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  background-color: #E0DFFF;
  padding: 12px 16px;
  border-radius: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  background-color: #C1EEDB;
  padding: 12px 16px;
  border-radius: 8px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  background-color: #FCD6C3;
  padding: 12px 16px;
  border-radius: 8px;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
    >
      Success banner without button
    </p>
  </div>
  <div
    class="c2"
  >
    <p
      class="c1"
    >
      Informative banner without button
    </p>
  </div>
  <div
    class="c3"
  >
    <p
      class="c1"
    >
      Attention banner without button
    </p>
  </div>
</div>
`;
